controllers/oscmachine_volume_controller_unit_test.go

```diff
@@ -0,0 +1,784 @@
+			clusterSpec:                       defaultClusterInitialize,
+			machineSpec:                       defaultVolumeInitialize,
+			expCheckVolumeFormatParametersErr: nil,
+		},
+		{
+			name:        "Check work without spec (with default values)",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{},
+			},
+			expCheckVolumeFormatParametersErr: nil,
+		},
+		{
+			name:        "Check Bad name volume",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume@test",
+							Iops:          1000,
+							Size:          50,
+							VolumeType:    "io1",
+							SubregionName: "eu-west-2a",
+						},
+					},
+				},
+			},
+			expCheckVolumeFormatParametersErr: fmt.Errorf("Invalid Tag Name"),
+		},
+		{
+			name:        "Check Bad Iops volume",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume",
+							Iops:          13001,
+							Size:          50,
+							VolumeType:    "io1",
+							SubregionName: "eu-west-2a",
+						},
+					},
+				},
+			},
+			expCheckVolumeFormatParametersErr: fmt.Errorf("Invalid iops"),
+		},
+		{
+			name:        "Check Bad size volume",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume",
+							Iops:          1000,
+							Size:          14902,
+							VolumeType:    "io1",
+							SubregionName: "eu-west-2a",
+						},
+					},
+				},
+			},
+			expCheckVolumeFormatParametersErr: fmt.Errorf("Invalid size"),
+		},
+		{
+			name:        "Check Bad SubregionName",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume",
+							Iops:          1000,
+							Size:          50,
+							VolumeType:    "io1",
+							SubregionName: "eu-west-2c",
+						},
+					},
+				},
+			},
+			expCheckVolumeFormatParametersErr: fmt.Errorf("Invalid subregionName"),
+		},
+		{
+			name:        "Check Bad volumeType",
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume",
+							Iops:          1000,
+							Size:          50,
+							VolumeType:    "gp3",
+							SubregionName: "eu-west-2a",
```

Comment:
Also check standard and gp2

Reply:
Check Bad volumeType
I use volume that does not exist.

Reply:
Yes i miss it